### DISCUSSION TITLE
[Feature] Transcode policy: accept open codecs (lpcm, flac, vp8)

### DISCUSSION THREAD

**Discussion by pyorot** [immich-app/immich#10540] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It's caused me problems (excessive storage usage) that I've been unable to disable the transcoding of files that open normally in my clients, that is, ones that use fairly universal open formats. As part of making transcode policy more flexible, we can add more options to the lists of accepted formats. We don't need to offer the option to transcode *into* them, but generally speaking, I would like to see a more extensive lists of codecs and a doc page talking loosely about what is expected to work where.

One could argue accepting vp8 and flac is unnecessary insofar as there aren't any devices known to capture photos/videos in them, but lpcm is both universal on the web (the unique uncompressed audio format, stored in wav container usually) and used by iPhones as part of the mov part of live photos.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by kevincox** [comment]
I'm especially interested in flac as it is a web-safe codec. It is supported in every modern browser: https://caniuse.com/flac

I have "original" files encoded with AV1 + flac that I would prefer not to generate thumbnails for. I just want to be able to add it to the `acceptedAudioCodecs` list.

Should be a fairly easy change to add it to the enum.

**Reply by pyorot** [reply]
I just got lpcm acceptance merged today so you can model a pull request off that and see the discussion and issues encountered there at #13418 👍.

Mostly amounts to running ffprobe on a representative file and adding the audio codec_name to the enums.
